hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -139,43 +116,43 @@ def _compute_due_hours(self):
                     if leave.state not in ['validate', 'validate1']:
                         continue
                     if not leave.holiday_status_id.remove_from_due_hours:
-                        # todo: use config instead of 8.0
+                        working_day = float(self.env[
+                            'ir.config_parameter'].get_param(
+                            'hr_attendance_calendar.work_day_duration'))
                         due_hours -= leave.compute_work_day(
-                            att_day.date) * 8
+                            att_day.date) * working_day
 
             att_day.due_hours = due_hours
 
     @api.multi
     @api.depends('attendance_ids.worked_hours')
     def _compute_worked_hours(self):
         for att_day in self.filtered('attendance_ids'):
-            total_hours = 0
-            for attendance in att_day.attendance_ids:
-                total_hours += attendance.worked_hours
-            att_day.worked_hours = total_hours
+            att_day.logged_hours = sum(
+                att_day.attendance_ids.mapped('worked_hours'))
 
     @api.multi
-    @api.depends('worked_hours')
+    @api.depends('logged_hours')
     def _compute_logged_hours(self):
-        # todo: at verify if it's still useful
-        pass
-
-    # @api.multi
-    # @api.depends('break_ids')
-    # def _compute_due_break(self):
-    #     for att_day in self:
-    #         more_hours = att_day.worked_hours > att_day.due_hours
-    #         hours = att_day.worked_hours if more_hours else att_day.due_hours
-    #         # todo: verify
-    #         rule = self.env['hr.attendance.rules'].search([
-    #             ('threshold', '>', hours),
-    #         ])
-    #         if rule:
-    #             att_day.due_break_min = rule[0].due_break
-    #             att_day.due_break_total = rule[0].due_break_total
-    #         else:
-    #             att_day.due_break_min = 0
-    #             att_day.due_break_total = 0
+        for att_day in self.filtered('coefficient_id'):
+            coefficient = att_day.coefficient_id.coefficient
+            att_day.worked_hours = att_day.logged_hours * coefficient
+
+    @api.multi
+    def _compute_due_break(self):
+        for att_day in self:
```

Comment:
I would add a comment in this method to explain a bit with human words the process, because this is hard to follow